### PR TITLE
Added files end point to vault (CADC-12561)

### PR DESCRIPTION
**Pull request by andamian**

### CODE REVIEW

File: cadc-vos-server/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ sourceCompatibility = 1.8
 
 group = 'org.opencadc'
 
-version = '2.0.9'
+version = '2.0.10'
```
Comment (pdowler) [inline cadc-vos-server/build.gradle:19]: currently published version is 2.0.8 so no change needed here

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java [hunk 2]
```diff
@@ -199,6 +205,39 @@ protected NodeWriter getNodeWriter() throws Exception {
         return ret;
     }
 
+    protected void setResponseHeaders(DataNode node) {
```
Comment (pdowler) [inline cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java:208]: I don't see why NodeAction is setting DataNode response headers. Explain?

File: cavern/src/main/java/org/opencadc/cavern/files/FileAction.java [hunk 3]
```diff
@@ -83,22 +83,21 @@
...
+import org.opencadc.vospace.server.actions.NodeAction;
 import org.opencadc.vospace.server.auth.VOSpaceAuthorizer;
 
 /**
  *
  * @author majorb
  * @author jeevesh
  */
-public abstract class FileAction extends RestAction {
+public abstract class FileAction extends NodeAction {
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/files/FileAction.java:94]: No - A FilesAction is not a NodeAction and should not subclass just because they happen to have some code in common.

**Review by pdowler — CHANGES_REQUESTED** [review]
I don't like using subclassing to slightly optimise code for actions on different endpoints. It ends up being hard to maintain... better for the "files" code in cavern to be independent of "nodes" code (from the lib). Same will apply to vault.

File: cavern/src/main/java/org/opencadc/cavern/files/FileAction.java (hunk 3 above)
Comment (andamian) [reply cavern/src/main/java/org/opencadc/cavern/files/FileAction.java]: Agree.

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java [hunk 4]
```diff
@@ -131,30 +137,10 @@ public void initAction() throws Exception {
     }
     
     @Override
-    protected final InlineContentHandler getInlineContentHandler() {
+    protected InlineContentHandler getInlineContentHandler() {
         return new InlineNodeHandler(INLINE_CONTENT_TAG);
     }
-    
-    // the absolute URI constructed from the request path
-    protected final VOSURI getTargetURI() {
```
Comment (pdowler) [inline cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java:139]: why move this to Utils and make all thiose other little changes? it doesn't seem related

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java [hunk 5]
```diff
@@ -131,30 +137,10 @@ public void initAction() throws Exception {
     }
     
     @Override
-    protected final InlineContentHandler getInlineContentHandler() {
```
Comment (pdowler) [inline cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java:134]: this should still be final

File: cavern/src/main/java/org/opencadc/cavern/files/FileAction.java [hunk 6]
```diff
@@ -83,22 +84,22 @@
...
+import org.opencadc.vospace.server.actions.NodeAction;
 import org.opencadc.vospace.server.auth.VOSpaceAuthorizer;
 
 /**
  *
  * @author majorb
  * @author jeevesh
  */
-public abstract class FileAction extends RestAction {
+public abstract class FileAction extends BaseFileAction {
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/files/FileAction.java:96]: revert most code changes in this cavern package; just keep the minimum required by API change in TransferGenerator

**Review by pdowler — CHANGES_REQUESTED** [review]

File: cadc-test-vos/src/main/java/org/opencadc/conformance/vos/FilesTest.java [hunk 7]
```diff
@@ -103,89 +113,173 @@ protected FilesTest(URI resourceID, File testCert) {
...
-            log.debug("file content: " + actual);
-            Assert.assertEquals("expected file content to match", expected, actual);
-
-            // Delete the node
-            delete(nodeURL);
-
-        } catch (Exception e) {
-            log.error("Unexpected error", e);
-            Assert.fail("Unexpected error: " + e);
+    public void fileTest() throws Exception {
```
[the remark this answers is not in the document]
Comment (andamian) [reply cadc-test-vos/src/main/java/org/opencadc/conformance/vos/FilesTest.java]: Reverted for clarity. The `try-catch` style doesn't do anything except shifting the code to the right.

File: cadc-test-vos/src/main/java/org/opencadc/conformance/vos/NodesTest.java [hunk 8]
```diff
@@ -307,6 +308,19 @@ public void testDataNode() {
                 Assert.fail("immutable prop test: " + len);
             }
 
+            // test view=data
```
[the remark this answers is not in the document]
Comment (andamian) [reply cadc-test-vos/src/main/java/org/opencadc/conformance/vos/NodesTest.java]: Didn't realized that. It would have been more obvious of the entire method was commented out.

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/BaseFileAction.java [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+import org.opencadc.vospace.server.LocalServiceURI;
+import org.opencadc.vospace.server.NodePersistence;
+import org.opencadc.vospace.server.Utils;
+import org.opencadc.vospace.server.auth.VOSpaceAuthorizer;
+
+/**
+ * Common functionality of an action on files end point.
+ * @author adriand
+ */
+public abstract class BaseFileAction extends RestAction {
```
[the remark this answers is not in the document]
Comment (andamian) [reply cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/BaseFileAction.java]: Reverted to the original plan.

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java (hunk 4 above)
Comment (andamian) [reply cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/NodeAction.java]: Reverted. Found a replacement of this method for `vault`

File: cavern/src/main/java/org/opencadc/cavern/files/FileAction.java (hunk 6 above)
Comment (andamian) [reply cavern/src/main/java/org/opencadc/cavern/files/FileAction.java]: Done

**Review by pdowler — APPROVED** [review]
